### PR TITLE
feat(hive): implement sync reader for hive table.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

### CODE REVIEW

File: src/query/storages/hive/hive/src/hive_parquet_block_reader.rs [hunk 1]
```diff
@@ -167,7 +167,7 @@ impl HiveParquetBlockReader {
     }
 
     async fn read_column(
-        o: Object,
+        o: Arc<Object>,
```
Comment (Xuanwo) [inline src/query/storages/hive/hive/src/hive_parquet_block_reader.rs:170]: Not needed, `Object` is cheap to clone.

Comment (RinChanNOWWW) [reply src/query/storages/hive/hive/src/hive_parquet_block_reader.rs:170]: Will cloned objects share the same fd?

File: src/query/storages/hive/hive/src/hive_parquet_block_reader.rs [hunk 2]
```diff
@@ -191,6 +191,11 @@ impl HiveParquetBlockReader {
         }
     }
 
+    fn sync_read_column(o: &Object, offset: u64, length: u64) -> Result<Vec<u8>> {
+        let chunk = o.blocking_range_read(offset..offset + length)?;
```
Comment (sundy-li) [inline src/query/storages/hive/hive/src/hive_parquet_block_reader.rs:195]: need to have it in opendal

Comment (Xuanwo) [reply src/query/storages/hive/hive/src/hive_parquet_block_reader.rs:195]: After https://github.com/datafuselabs/opendal/pull/1085 implemented, we can use `read_at` for better performance.

File: src/query/storages/hive/hive/src/hive_parquet_block_reader.rs [hunk 3]
```diff
@@ -266,4 +291,8 @@ impl HiveBlockReader {
             .next_block(&self.projected_schema, &self.hive_partition_filler, &part)
             .map_err(|e| e.add_message(format!(" filename of hive part {}", part.filename)))
     }
+
+    pub fn support_blocking_api(&self) -> bool {
+        self.operator.metadata().can_blocking()
```
Comment (sundy-li) [inline src/query/storages/hive/hive/src/hive_parquet_block_reader.rs:296]: seems we need another flag to control them.

can we have a setting named `hive_source_blocking_read_mode` ?

Comment (RinChanNOWWW) [reply src/query/storages/hive/hive/src/hive_parquet_block_reader.rs:296]: Databend-query setting?

**Comment by Xuanwo** [comment]
OpenDAL 0.23 has been released: https://github.com/datafuselabs/opendal/discussions/1096

blocking operation has been implemented on hdfs.

**Comment by RinChanNOWWW** [comment]
Is this feature still needed?

**Comment by sandflee** [comment]
> Is this feature still needed?

this pr contains 2 part:
1,  serial read column datas,  this's useful in some conditions, but should use object handler to do `fd reuse` and add a setting to enable it?
2,  put hdfs op to pipeline sync stage,  it does't help too much ,  @Xuanwo is working on put hdfs to blocking thread pool.

**Comment by sundy-li** [comment]
Let's close it and @Xuanwo  will improve it another way.

**Pull request closed without merging**